flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -231,6 +231,11 @@ public Builder writeParallelism(int newWriteParallelism) {
       return this;
     }
 
+    public Builder properties(Map<String, String> properties) {
+      upsert(PropertyUtil.propertyAsBoolean(properties, UPSERT_ENABLED, UPSERT_ENABLED_DEFAULT));
```

Comment:
Nit: Please add a `Preconditions` check that the map isn't null (unless null needs to be allowed).